modules/indexer/code/elastic_search.go

```diff
@@ -251,6 +252,36 @@ func (b *ElasticSearchIndexer) Delete(repoID int64) error {
 	return err
 }
 
+// indexPos find words positions for start and the following end on content. It will
+// return the beginning position of the first start and the ending position of the
+// first end following the start string.
+// If not found any of the positions, it will return -1, -1.
+func indexPos(content, start, end string) (int, int) {
+	if len(content) < len(start)+len(end) {
+		return -1, -1
+	}
+	startIdx, endIdx := -1, -1
+	for i := 0; i < len(content); i++ {
+		if startIdx > -1 {
+			if i+len(end) > len(content) {
+				return -1, -1
+			}
+			if content[i:i+len(end)] == end {
+				endIdx = i
+				return startIdx, endIdx + len(end)
+			}
+		} else {
+			if i+len(start) > len(content) {
+				return -1, -1
+			}
+			if content[i:i+len(start)] == start {
+				startIdx = i
+			}
+		}
+	}
+	return -1, -1
+}
```

Comment:
So strings.Index(...) is actually quicker than iterating through the indices 

```suggestion
func indexPos(content, start, end string) (int, int) {
	if len(content) < len(start)+len(end) {
		return -1, -1
	}
	startIdx := strings.Index(content, start)
	if startIdx < 0 {
		return -1, -1
	}
	pos := startIdx + len(start)
	endIdx := strings.Index(content[pos:], end)
	if endIdx < 0 {
		return -1, -1
	}
	
	return startIdx, pos+endIdx
}
```